uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Helper/NavigationMenuHelper.php

```diff
@@ -25,24 +25,30 @@ public static function findMenuItemByPath($menu, $path) {
    *
    * @param array $menu
    *   menu hierarchy
-   * @param string $parentPath
+   * @param string $path
    *   Path to the parent menu item, e.g. "Events/Dashboard"
-   * @param array $itemAttributes
-   *   Attributes of item to insert, e.g. ['name' => 'New Item']
+   * @param array $attributes
+   *   Attributes of item to insert, e.g. ['label' => 'New Item']
    */
-  public static function insertChild(&$menu, $parentPath, $itemAttributes) {
-    $parent = &self::findMenuItemReferenceByPath($menu, $parentPath);
-
-    if (!$parent) {
-      $err = sprintf('Cannot find parent item "%s"', $parentPath);
-      throw new \Exception($err);
-    }
+  public static function insert(&$menu, $path, $attributes = []) {
```

Comment:
These changes were to allow insertion of items into the top level (root) menu

Reply:
The description is what's wrong there. In the previous version the parent path was provided, like

`NavigationMenuHelper::insert($menu, 'Events', ['name' => 'Dashboard']);`

But now the name is inferred from the item you want to insert.

`NavigationMenuHelper::insert($menu, 'Events/Dashboard');`

Both of the above examples would insert an item called "Dashboard" into the "Events" submenu.

You can still override it using the optional attributes. The reason I changed it is that I found it easier to use this because I can focus only on the item I want to insert, and not the parent. If the parent doesn't exist it'll still throw an exception.

I can update the method description to match what it does now.